helmctlr/helm.go

```diff
@@ -71,13 +72,18 @@ func NewController(chartDir, ns, rn, host string, wait bool, waitto int64, logge
 func (c Controller) ResourceAdded(r *unstructured.Unstructured) {
 	metrics.TotalEvents.Inc()
 	c.logger.Infow("resource added", "resource", r.GetName())
-	if err := c.installOrUpdate(r); err != nil {
+	c.updateCRStatus(r, crw.PhaseApplying, crw.ReasonCustomResourceAdded, "resource added: " + r.GetName())
```

Comment:
there's no reason to set a status before we've tried to do anything if there's no way to return from this control loop before we try to do anything.

I think we can just remove this status update entirely

Reply:
@josephschorr suggested this status since the "apply" operation is synchronous.

Reply:
I didn't phrase that very well, sorry - 

What I mean is that there shouldn't be a path through this control loop that updates the CR twice. We'll process an update event for each time `updateCRStatus` is called, so if that's called twice within the same loop (ResourceAdded or ResourceUpdated) we'll get stale events in our loop.

One option is to exit as soon as the first status is set, and then check for that status before processing further.